opentracing-brave/src/main/java/io/opentracing/BraveSpanImpl.java

```diff
@@ -0,0 +1,78 @@
+/**
+ * Copyright 2016 The OpenTracing Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package io.opentracing;
+
+import java.time.Instant;
+import java.util.Optional;
+import java.util.concurrent.TimeUnit;
+
+import com.github.kristofa.brave.Brave;
+import com.github.kristofa.brave.ClientTracer;
+import com.github.kristofa.brave.ServerTracer;
+import com.github.kristofa.brave.SpanId;
+
+
+final class BraveSpanImpl extends AbstractSpan {
+
+    final SpanId spanId;
+    final Optional<Span> parent;
+    final Optional<ServerTracer> serverTracer;
```

Comment:
> What I noticed was that the act of injecting or extracting didn't seem to have any side-effects on the span …

To be honest I haven't thought about using the brave collector directly.

I've implemented this in mind with the platform being traced is being traced by Zipkin.
Therefore it makes sense to use local/client/server spans as is expected for Zipkin.
For example the platform might not be consistently using OT, but only using it in those components that are also deployed to other platforms that have different tracing backends.

I'm not sure that really contributes much though to the decision of using Brave's porcelain or plumbing api here. Otherwise AFAIK we're sticking with OT idioms here, while implementing Zipkin behaviour.

And I can't see how the idea of using the plumbing (the sampler and collector) isn't going to save any code or complexity here.

@adriancole